python-sdk/nuscenes/eval/detection/algo.py

```diff
@@ -114,142 +130,79 @@ def average_precision(gt_boxes: EvalBoxes,
-
-    # Store modified rec and prec values.
-    metrics['rec'] = rec
-    metrics['prec'] = prec
-
-    # If the max recall is below the minimum recall range, return the maximum error.
-    if max(rec) < min(score_range):
-        return np.nan, dict()
-
-    # Find indices of rec that are close to the interpolated recall thresholds.
-    assert all(rec == sorted(rec))  # np.searchsorted requires sorted inputs.
-    thresh_count = int((score_range[1] - score_range[0]) * 100 + 1)
-    rec_interp = np.linspace(score_range[0], score_range[1], thresh_count)
-    threshold_inds = np.searchsorted(rec, rec_interp, side='left').astype(np.float32)
-    threshold_inds[threshold_inds == len(rec)] = np.nan  # Mark unachieved recall values as such.
-    assert np.nanmax(threshold_inds) < len(rec)  # Check that threshold indices are not out of bounds.
-    metrics['threshold_inds'] = threshold_inds
-
-    # Interpolation of precisions to the nearest lower recall threshold.
-    # For unachieved high recall values the precision is set to 0.
-    prec_interp = np.interp(rec_interp, rec, prec, right=0)
-    metrics['rec_interp'] = rec_interp
-    metrics['prec_interp'] = prec_interp
-
-    # Compute average precision over predefined thresholds.
-    average_prec = prec_interp.mean()
-
-    # Plot PR curve.
-    # plt.clf()
-    # plt.plot(metrics['rec'], metrics['prec'])
-    # plt.plot(metrics['rec'], metrics['conf'])
-    # plt.xlabel('Recall')
-    # plt.ylabel('Precision (blue), Conf (orange)')
-    # plt.ylim([0.0, 1.05])
-    # plt.xlim([0.0, 1.0])
-    # plt.title('Precision-Recall curve: class %s, dist_th=%.1fm, AP=%.4f' % (class_name, dist_th, average_prec))
-    # save_path = os.path.join(self.plot_dir, '%s-recall-precision-%.1f.png' % (class_name, dist_th))
-    # plt.savefig(save_path)
-
-    # Print stats.
-    if verbose:
-        tp_count = tp[-1] if len(tp) > 0 else 0
-        fp_count = fp[-1] if len(fp) > 0 else 0
-        print('AP is %.4f, tp: %d, fp: %d' % (average_prec, tp_count, fp_count))
-
-    return average_prec, metrics
-
-
-def calc_tp_metrics(raw_metrics: Dict, cfg: DetectionConfig, class_name: str, verbose: bool) -> Dict[str, float]:
-    """
-    True Positive metrics for a single class and distance threshold.
-    For each metric and recall threshold we compute the mean for all matches with a lower recall.
-    The raw_metrics are computed in average_precision() to avoid redundant computation.
-    :param raw_metrics: Raw data for a number of metrics.
-    :param cfg: Config.
-    :param class_name: Class to compute AP on.
-    :param verbose: Whether to print outputs to console.
-    :return: Maps each metric name to its average metric error or nan to be ignored.
-    """
+        confs = np.append(1, confs)
+
+    # ---------------------------------------------
+    # Step3: Re-sample recall, precision and confidences such that we have one data point for each
+    # recall fraction between 1 and 100
```

Comment:
1 and 100 sounds confusing, as it's 1-indexed. Perhaps "for each percentage between 0 and 1" ?